### DISCUSSION TITLE
Is directory structure customization available in Community or Enterprise versions?

### DISCUSSION THREAD

**Discussion by SSaiPrasad7** [emqx/emqx#4296] [Q&A] [answered]
Hi,
Presently I am using Community version of EMQX server. I have a doubt whether we can provide our own custom directory for `.conf ` files instead of using existing directory structure( `/etc/emqx/plugins/`).
If the above problem statement is possible,so guide me to the solution.
Thank you.

**Comment by zmstone** [comment]
hi
@saiprasadsamudrala which version are you basing your developments on?

**Comment by SSaiPrasad7** [comment]
community version

**Reply by zmstone** [reply]
Do you mean the latest master branch?

**Reply by SSaiPrasad7** [reply]
Yes. EMQ X Broker version 4.2

**Reply by SSaiPrasad7** [reply]
@zmstone. Can you help me with the problem statement?

**Reply by zmstone** [reply]
Sure. I was just trying to make sure you are not using a too old version which I’m not quite familiar with.
Tried to answer in the other reply.

**Comment by zmstone** [comment, marked as answer]
if you want to reuse our plugins framework, there is no easy way to avoid having a conf file in etc/plugins dir.

However you should be able to read additional config files from anywhere in your plugin code.
E.g read and parse a config file in this function https://github.com/emqx/emqx-plugin-template/blob/8d0441fa55c66c3bad3b77fecf16a665a3ba1962/src/emqx_plugin_template_app.erl#L27

**Reply by SSaiPrasad7** [reply]
Thank you @zmstone
